### PR TITLE
[Build] remove mkdir for swift module paths since it is inferred by llbuild now.

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

File: Sources/Build/Command.compile(SwiftModule).swift [hunk 1]
```diff
@@ -35,7 +35,6 @@ extension Command {
 
         let tool = SwiftcTool(module: module, prefix: prefix, otherArgs: args + otherArgs, executable: SWIFT_EXEC, conf: conf)
 
-        //FIXME these should be inferred as implicit inputs by llbuild
         let mkdirs = Set(tool.objects.map{ $0.parentDirectory }).map(Command.createDirectory)
```
Comment (ddunbar) [inline Sources/Build/Command.compile(SwiftModule).swift]: Err, this isn't right. The FIXME is about not needing to run these commands at all, the right resolution to that issue is to ditch the createDirectory call and make sure stuff still works.

Comment (aciidb0mb3r) [inline Sources/Build/Command.compile(SwiftModule).swift]: Right. I might have run into a wall couple of mins back.

**Comment by aciidb0mb3r** [comment]
@ddunbar re-review

File: Sources/Build/Command.compile(SwiftModule).swift [hunk 2]
```diff
@@ -34,9 +30,6 @@ extension Command {
         #endif
 
         let tool = SwiftcTool(module: module, prefix: prefix, otherArgs: args + otherArgs, executable: SWIFT_EXEC, conf: conf)
-
-        //FIXME these should be inferred as implicit inputs by llbuild
-        let mkdirs = Set(tool.objects.map{ $0.parentDirectory }).map(Command.createDirectory)
-        return (cmd(tool), mkdirs)
```
Comment (ddunbar) [inline Sources/Build/Command.compile(SwiftModule).swift]: These probably shouldn't be combined in one commit, this is a subtle semantic change, the other is a trivial refactor.

Comment (aciidb0mb3r) [inline Sources/Build/Command.compile(SwiftModule).swift]: Separated into two commits

**Comment by ddunbar** [comment]
LGTM, thanks this had been on my TODO list!

**Comment by ddunbar** [comment]
@swift-ci please test and merge

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
I removed mkdir from clang module too, now `Command.createDirectory` isn't used anywhere. Should we remove it or let it stay for future needs?

**Comment by aciidb0mb3r** [comment]
Hmm this feels like CI malfunction, it shouldn't have merged it since my command was 'test' and the previous 'test and merge' job was cancelled when I pushed new commits. /cc @shahmishal @tkremenek

**Comment by shahmishal** [comment]
@aciidb0mb3r Looking into the issue, CI system is not setup to stop merge job if new commit has been introduced after triggering test and merge.

**Pull request merged**